plugins/zstack.py

```diff
@@ -0,0 +1,376 @@
+        if hasattr(tab_data, "spotStream"):
+            try:
+                ss.remove(tab_data.spotStream)
+            except ValueError:
+                pass  # spotStream was not there anyway
+
+        for s in ss:
+            if isinstance(s, StaticStream):
+                ss.remove(s)
+        return ss
+
+    def _get_acq_streams(self):
+        """
+        Return the streams that should be used for acquisition
+        return:
+           acq_st (list of streams): the streams to be acquired at every repetition
+           last_st (list of streamsintp): streams to be acquired at the end
+        """
+        if not self._dlg:
+            return []
+
+        live_st = (self._dlg.microscope_view.getStreams() +
+                   self._dlg.hidden_view.getStreams())
+        logging.debug("View has %d streams", len(live_st))
+
+        # On the SPARC, the acquisition streams are not the same as the live
+        # streams. On the SECOM/DELPHI, they are the same (for now)
+        tab_data = self.main_app.main_data.tab.value.tab_data_model
+        if hasattr(tab_data, "acquisitionStreams"):
+            acq_st = tab_data.acquisitionStreams
+            # Discard the acquisition streams which are not visible
+            ss = []
+            for acs in acq_st:
+                if isinstance(acs, stream.MultipleDetectorStream):
+                    if any(subs in live_st for subs in acs.streams):
+                        ss.append(acs)
+                        break
+                elif acs in live_st:
+                    ss.append(acs)
+        else:
+            # No special acquisition streams
+            ss = live_st
+
+        last_ss = []
+        self._acq_streams = acq.foldStreams(ss, self._acq_streams)
+        return self._acq_streams, last_ss
+
+    def start(self):
+        # Fail if the live tab is not selected
+        tab = self.main_app.main_data.tab.value
+        if tab.name not in ("secom_live", "sparc_acqui"):
+            box = wx.MessageDialog(self.main_app.main_frame,
+                       "ZStack acquisition must be done from the acquisition stream.",
+                       "ZStack acquisition not possible", wx.OK | wx.ICON_STOP)
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        # On SPARC, fail if no ROI selected
+        try:
+            if tab.tab_data_model.semStream.roi.value == UNDEFINED_ROI:
+                box = wx.MessageDialog(self.main_app.main_frame,
+                           "You need to select a region of acquisition.",
+                           "Z stack acquisition not possible", wx.OK | wx.ICON_STOP)
+                box.ShowModal()
+                box.Destroy()
+                return
+        except AttributeError:
+            pass  # Not a SPARC
+
+        # Stop the stream(s) playing to not interfere with the acquisition
+        tab.streambar_controller.pauseStreams()
+
+        self.filename.value = self._get_new_filename()
```

Comment:
Please set zstart to the current focus level.

Ideally, it should be possible to change the focus on the view, and have a "set" button next to the "Z-start" to update the value. It would also have a Z-stop to be able to also set the end. Depending on which VA is changed, the other VA (eg, Z-step) would automatically update.

See: http://www.micro-manager.org/w/images/6/64/MM_multid.jpg